akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -886,7 +910,13 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       // removing REMOVED nodes from the `reachability` table
       val newReachability = localOverview.reachability.remove(removed)
       val newOverview = localOverview copy (seen = newSeen, reachability = newReachability)
-      val newGossip = localGossip copy (members = newMembers, overview = newOverview)
+      val newVersion = removed.foldLeft(localGossip.version) { (v, node) ⇒
+        v.prune(VectorClock.Node(vclockName(node)))
+      }
+      val newGossip = localGossip copy (members = newMembers, overview = newOverview, version = newVersion)
```

Comment:
The idea (if I remember correctly) is that the change made by the leader is stamped by the leader node (incrementing its slot in the vclock). That will propagate as is if there are no other changes on other nodes.

If there are changes on other nodes, the merge logic kicks in. (I'll have to revive my memories of that part)

If there are changes from the removed node itself we might have a minor problem. That change might be lost. On the other hand we don't allow any changes from a leaving node (e.g. joining is avoided).